### PR TITLE
Add docs for model code

### PR DESCRIPTION
**Pull request by pyeres**
This PR includes draft docstrings for some model code. Exposing these docs for feedback. 

(+ @HaokunLiu for visibility)

### CODE REVIEW

File: jiant/proj/main/modeling/model_setup.py [hunk 1]
```diff
@@ -340,6 +357,16 @@ def get_model_arch_from_jiant_model(jiant_model: nn.Module) -> ModelArchitecture
 
 
 def get_ancestor_model(transformers_class_spec, model_config_path):
+    """Load the model config from a file, configure the model, and return the model.
```
Comment (zphang) [inline jiant/proj/main/modeling/model_setup.py:360]: Clarify that the model here, in the case of Transformer models, corresponds to the model class that contains all of the pretrained weights (e.g. both MLM and NSP heads)

File: jiant/proj/main/modeling/model_setup.py [hunk 2]
```diff
@@ -262,6 +262,23 @@ def create_taskmodel(
 
 
 def get_encoder(model_arch, ancestor_model):
+    """From model architecture, get the bare model (outputting raw hidden-states without a head).
```
Comment (zphang) [inline jiant/proj/main/modeling/model_setup.py:265]: Generally, encoder = embedding layer + self-attention layers

File: jiant/shared/model_resolution.py [hunk 3]
```diff
@@ -199,4 +208,13 @@ def build_featurization_spec(model_type, max_seq_length):
 
 
 def resolve_tokenizer_class(model_type):
+    """Get tokenizer for a given model architecture.
```
Comment (zphang) [inline jiant/shared/model_resolution.py:211]: Clarify: tokenizer *class*. The tokenizer is still not yet instantiated / loaded.

Comment (pyeres) [reply jiant/shared/model_resolution.py]: Thanks, done w/ 64b5a99.

File: jiant/proj/main/modeling/model_setup.py (hunk 2 above)
Comment (pyeres) [reply jiant/proj/main/modeling/model_setup.py]: Thanks, done w/ 64b5a99.

File: jiant/proj/main/modeling/model_setup.py (hunk 1 above)
Comment (pyeres) [reply jiant/proj/main/modeling/model_setup.py:360]: Thanks, done w/ 64b5a99.

**Pull request merged**